client/classic/upe/payment-processing.js

```diff
@@ -92,9 +95,42 @@ function createStripePaymentElement( api, paymentMethodType = null ) {
 	gatewayUPEComponents[
 		paymentMethodType
 	].upeElement = createdStripePaymentElement;
+
+	// When email or phone is updated and Link is enabled, we need to
+	// update the payment element to update its default values.
+	if (
+		getStripeServerData()?.isCheckout &&
+		isLinkEnabled() &&
+		paymentMethodType === 'card'
+	) {
+		if ( document.getElementById( 'billing_email' ) ) {
```

Comment:
Optional: Maybe we could have a specific function to avoid duplicate code? Something like:
```
function maybeAttachPaymentElementDefaultValueUpdateToBlur( elementId ) {
	if ( ! document.getElementById( elementId ) ) {
		return;
	}

	document.getElementById( elementId ).onblur = function () {
		updatePaymentElementDefaultValues();
	};
}
```
Not the best name. And I know it is just one almost-duplicated line, but these things tend to grow fast.

Reply:
Good suggestion! Tidied it up in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3621/commits/e2572e0bed04ade9c6e8056979a5e0985c477df1